RInAppMessaging/Classes/InAppMessagingModule.swift

```diff
@@ -105,10 +105,10 @@ internal class InAppMessagingModule: AnalyticsBroadcaster,
 
         let isLogoutOrUserChange = (preferenceRepository.getUserIdentifiers().isEmpty || diff?.isEmpty == false) && !oldUserIdentifiers.isEmpty
         if isLogoutOrUserChange {
-            campaignRepository.resetDataPersistence()
+            campaignRepository.clearLastUserData()
             eventMatcher.clearNonPersistentEvents()
         }
-        campaignRepository.loadCachedData(syncWithLastUserData: false)
+        campaignRepository.loadCachedData(syncWithLastUserData: true)
```

Comment:
Was this wrong before?
Why change it to true?

Reply:
I think it was kind of wrong before. It could cause problems in some cases. This part covers the 'limbo' state when SDK is initialized and receiving events but user preference is not yet registered. This also fulfils `will transfer cached data (sync) from empty user` test.